library/src-bootstrapped/scala/runtime/Tuple.scala

```diff
@@ -463,4 +463,38 @@ object Tuple {
   def concatIterator(tup1: Tuple, tup2: Tuple): Iterator[Any] =
     tup1.asInstanceOf[Product].productIterator ++ tup2.asInstanceOf[Product].productIterator
 
+  def isInstanceOfTuple(x: Any): Boolean =
+    x.isInstanceOf[Product] && isInstanceOfEmptyTuple(x) || isInstanceOfNonEmptyTuple(x)
+
+  def isInstanceOfEmptyTuple(x: Any): Boolean = x == Tuple0
+
+  def isInstanceOfNonEmptyTuple(x: Any): Boolean = isInstanceOfPair(x)
+
+  def isInstanceOfPair(x: Any): Boolean =
+    x.isInstanceOf[Product] && (
```

Comment:
Since it will probably be common to call this on an EmptyTuple, an extra check to avoid the tests below could be worth it:
```suggestion
    x.isInstanceOf[Product] && !x.isInstanceOf[EmptyTuple] (
```

Reply:
Won't be needed with the switch